src/main/java/fi/aalto/cs/intellij/services/CourseInformation.java

```diff
@@ -0,0 +1,106 @@
+package fi.aalto.cs.intellij.services;
+
+import com.intellij.openapi.components.ServiceManager;
+
+import java.io.FileNotFoundException;
+import java.io.FileReader;
+import java.net.MalformedURLException;
+import java.net.URL;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.Iterator;
+import java.util.List;
+import java.util.Map;
+
+import org.json.JSONException;
+import org.json.JSONObject;
+import org.json.JSONTokener;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class CourseInformation {
+  private static final Logger logger = LoggerFactory
+      .getLogger(CourseInformation.class);
+
+  private String courseName;
+  private Map<String, URL> moduleUrls;
+
+  public CourseInformation() {
+    // Replace this with the correct path when testing with a local course configuration file.
+    parse("o1.json");
+  }
+
+  /**
+   * The CourseInformation instance managed by the service manager.
+   */
+  public static CourseInformation getInstance() {
+    return ServiceManager.getService(CourseInformation.class);
+  }
+
+  /**
+   * Parses the given course configuration file.
+   * @param pathToCourseConfig The path to the course configuration file.
+   */
+  public void parse(String pathToCourseConfig) {
```

Comment:
consider splitting this into multiple smaller methods

Reply:
after longish discussions, the approach we currently gravitate to: normal functions with proper inputs and outputs and as little as possible class variables (only for the special cases)

Reply:
Good point, probably a better idea to let this method throw exceptions, although the question of where to catch them is somewhat complicated by the current service manager approach.